frontend/packages/console-plugin-sdk/src/useExtensions.ts

```diff
@@ -40,47 +34,48 @@ import { Extension, ExtensionTypeGuard, LoadedExtension } from './typings';
  * };
  * ```
  *
- * @param typeGuard Type guard used to narrow the extension type.
+ * @param typeGuards Type guard(s) used to narrow the extension instances.
+ *
+ * @returns List of extension instances which are currently in use, narrowed by the
+ * given type guard(s).
  */
 export const useExtensions = <E extends Extension>(
-  typeGuard: ExtensionTypeGuard<E>,
+  ...typeGuards: ExtensionTypeGuard<E>[]
 ): LoadedExtension<E>[] => {
-  const allExtensions = pluginStore.getAllExtensions();
+  if (typeGuards.length === 0) {
+    throw new Error('You must pass at least one type guard to useExtensions');
+  }
+
+  const unsubscribeRef = React.useRef<VoidFunction>(null);
+  const extensionsInUseRef = React.useRef<E[]>([]);
+  const latestTypeGuardsRef = React.useRef<ExtensionTypeGuard<E>[]>(typeGuards);
+  const forceRender = React.useReducer((s: boolean) => !s, false)[1] as VoidFunction;
+
+  const trySubscribe = React.useCallback(() => {
+    if (unsubscribeRef.current === null) {
+      unsubscribeRef.current = subscribeToExtensions<E>((extensions) => {
+        extensionsInUseRef.current = extensions;
+        forceRender();
+      }, ...latestTypeGuardsRef.current);
+    }
+  }, [forceRender]);
+
+  const tryUnsubscribe = React.useCallback(() => {
+    if (unsubscribeRef.current !== null) {
+      unsubscribeRef.current();
+      unsubscribeRef.current = null;
+    }
+  }, []);
 
-  // 1) Narrow extensions according to type guard
-  const matchedExtensions = React.useMemo(() => allExtensions.filter(typeGuard), [
-    allExtensions,
-    typeGuard,
-  ]);
+  if (!_.isEqual(latestTypeGuardsRef.current, typeGuards)) {
+    latestTypeGuardsRef.current = typeGuards;
+    tryUnsubscribe();
+    trySubscribe();
```

Comment:
Unnecessary because this function is called immediate after the `if` block
```suggestion
```

Reply:
Right, I fixed this in follow-up #6655 but will do the change here as suggested.